x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/TransportVersionClusterStateUpgradeIT.java

```diff
@@ -154,12 +155,17 @@ private boolean runTransportVersionsTest() throws Exception {
                         logger.info("{} - not fixed up yet, retrying", nodeDescription);
                         return false;
                     }
-                } else if (nodeVersion.after(VERSION_INTRODUCING_TRANSPORT_VERSIONS)) {
-                    // There's no relationship between node versions and transport versions any more, although we can be sure of this:
-                    assertThat(nodeDescription, transportVersion, greaterThan(FIRST_TRANSPORT_VERSION));
                 } else {
-                    // Responding node is not upgraded, and no later than 8.8.0, so we infer its version correctly.
-                    assertEquals(nodeDescription, TransportVersion.fromId(nodeVersion.id()), transportVersion);
+                    var version = parseLegacyVersion(nodeVersion);
+                    // All non-semantic versions are after 8.8.0 and have transport version
+                    var transportVersionIntroduced = version.map(v -> v.after(VERSION_INTRODUCING_TRANSPORT_VERSIONS)).orElse(true);
```

Comment:
This is the same as `ClusterState.INFERRED_TRANSPORT_VERSION`, could you just directly reference that?

Reply:
Nice, will do! So VERSION_INTRODUCING_TRANSPORT_VERSIONS can be removed

Reply:
It may be worth renaming the constant in `ClusterState` to emphasise its wider usage